chia-wallet/src/proof.rs

```diff
@@ -1,26 +1,25 @@
 use arbitrary::{Arbitrary, Unstructured};
-use clvm_traits::{FromClvm, Result, ToClvm};
-use clvmr::{allocator::NodePtr, Allocator};
+use clvm_traits::{ClvmDecoder, ClvmEncoder, FromClvm, FromClvmError, ToClvm, ToClvmError};
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Proof {
     Lineage(LineageProof),
     Eve(EveProof),
 }
 
-impl FromClvm for Proof {
-    fn from_clvm(a: &Allocator, node: NodePtr) -> Result<Self> {
-        LineageProof::from_clvm(a, node)
+impl<N> FromClvm<N> for Proof {
+    fn from_clvm(decoder: &impl ClvmDecoder<Node = N>, node: N) -> Result<Self, FromClvmError> {
+        LineageProof::from_clvm(decoder, decoder.clone_node(&node))
```

Comment:
is the point of the `clone_node()` call to avoid having to specify the trait bounds for `N` as `Clone`?

Reply:
Yep, unfortunate that the compiler can't infer this, since it's required by `type Node: Clone` in Encoder and Decoder anyway.